client/my-sites/plans/jetpack-plans/product-store/utils/get-product-icon.ts

```diff
@@ -0,0 +1,37 @@
+import {
+	JETPACK_ANTI_SPAM_PRODUCTS,
+	JETPACK_BACKUP_PRODUCTS,
+	JETPACK_BOOST_PRODUCTS,
+	JETPACK_CRM_PRODUCTS,
+	JETPACK_SCAN_PRODUCTS,
+	JETPACK_SEARCH_PRODUCTS,
+	JETPACK_VIDEOPRESS_PRODUCTS,
+} from '@automattic/calypso-products';
+import JetpackProductIconAntiSpam from 'calypso/assets/images/jetpack/jetpack-product-icon-antispam.svg';
+import JetpackProductIconBackup from 'calypso/assets/images/jetpack/jetpack-product-icon-backup.svg';
+import JetpackProductIconBoost from 'calypso/assets/images/jetpack/jetpack-product-icon-boost.svg';
+import JetpackProductIconCRM from 'calypso/assets/images/jetpack/jetpack-product-icon-crm.svg';
+import JetpackProductIconScan from 'calypso/assets/images/jetpack/jetpack-product-icon-scan.svg';
+import JetpackProductIconSearch from 'calypso/assets/images/jetpack/jetpack-product-icon-search.svg';
+import JetpackProductIconVideopress from 'calypso/assets/images/jetpack/jetpack-product-icon-videopress.svg';
+import { productIconProps } from '../types';
+
+const setProductsIcon = ( slugs: ReadonlyArray< string >, src: string ) =>
+	slugs.reduce( ( map, slug ) => ( { ...map, [ slug ]: src } ), {} );
+
+const PRODUCT_ICON_MAP: Record< string, string > = {
+	...setProductsIcon( JETPACK_BACKUP_PRODUCTS, JetpackProductIconBackup ),
+	...setProductsIcon( JETPACK_ANTI_SPAM_PRODUCTS, JetpackProductIconAntiSpam ),
+	...setProductsIcon( JETPACK_SCAN_PRODUCTS, JetpackProductIconScan ),
+	...setProductsIcon( JETPACK_VIDEOPRESS_PRODUCTS, JetpackProductIconVideopress ),
+	...setProductsIcon( JETPACK_SEARCH_PRODUCTS, JetpackProductIconSearch ),
+	...setProductsIcon( JETPACK_BOOST_PRODUCTS, JetpackProductIconBoost ),
+	// TODO - Set Jetpack social product icon here
+	...setProductsIcon( JETPACK_CRM_PRODUCTS, JetpackProductIconCRM ),
+};
```

Comment:
Is there a reason for not doing simply this?

```js
const PRODUCT_ICON_MAP: Record< string, string > = {
	[ JETPACK_BACKUP_PRODUCTS ]: JetpackProductIconBackup,
	// other iterms
}
```

Reply:
Hi @manzoorwanijk, right now the `getProductIcon` accepts specific product slug like `jetpack_backup_t1_yearly` which is contained  at `JETPACK_BACKUP_PRODUCTS`.  I copied the same technique in the `product-grid/products-order.ts` to map all product slugs by creating a helper function so I don't have to repeat some codes like below:

```
const PRODUCT_ICON_MAP: Record< string, string > = {
  ...
  [ JETPACK_BACKUP_PRODUCTS_YEARLY ]: JetpackProductIconBackup,
  [ JETPACK_BACKUP_PRODUCTS_MONTHLY ]: JetpackProductIconBackup,
  ...
};
```